crates/ide-completion/src/completions/type.rs

```diff
@@ -157,7 +173,7 @@ pub(crate) fn complete_type_path(
                     });
                     return;
                 }
-                TypeLocation::GenericArgList(Some(arg_list)) => {
+                TypeLocation::GenericArgList(Some((arg_list, _))) => {
```

Comment:
The code below here is specifically looking for an `AssocTypeArg`, so we can simplify this now with the added resolution

Reply:
I simplified it somewhat by passing an `Option<hir::Trait>` to `TypeLocation::GenericArg` and separating `<Foo = $0>` type positions into new `TypeLocation::AssocTypeEq` and `TypeLocation::AssocConstEq` variants.